manifests/init.pp

```diff
@@ -0,0 +1,57 @@
+# Define advanced_audit_policies
+define advanced_audit_policy (
+  String                    $policy  = $title,
+  Enum['present', 'absent'] $ensure  = 'present',
+  Enum['enable', 'disable'] $success = 'disable',
+  Enum['enable', 'disable'] $failure = 'disable',
+) {
+
+  if "${::operatingsystem} ${::operatingsystemmajrelease}" !~ /(w|W)indows (2008( R2)?|2012( R2)?|2016|10)/ {
```

Comment:
user $facts['os']['family'] and so, maybe use unless, more fan of unless in these cases ;)